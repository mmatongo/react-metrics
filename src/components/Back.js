import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';

const Back = ({ title }) => {
  const history = useHistory();
  const back = () => {
    history.push('/');
  };

  return (
    <button type="button" onClick={back}>{title}</button>
  );
};

Back.propTypes = ({
  title: PropTypes.string,
}).isRequired;

export default Back;
