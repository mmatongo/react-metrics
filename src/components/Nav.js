import React from 'react';
import PropTypes from 'prop-types';

const Nav = ({ backButton, title }) => (
  <div className="nav-container">
    <div className="nav-content">
      <p>{backButton}</p>
      <p>{title}</p>
      <i>
        Thing 1
      </i>
      <br />
      <i>
        Thing 2
      </i>
    </div>
  </div>
);

Nav.propTypes = ({
  title: PropTypes.string,
  backButton: PropTypes.string,
}).isRequired;

export default Nav;
