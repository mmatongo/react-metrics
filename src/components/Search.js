import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ value, handler }) => (
  <div className="search">
    {' Search by Death count: '}
    <input
      type="number"
      id="deaths"
      name="deaths"
      value={value}
      onChange={handler}
    />
  </div>
);

Search.propTypes = ({
  value: PropTypes.number,
  handler: PropTypes.func,
}).isRequired;

export default Search;
