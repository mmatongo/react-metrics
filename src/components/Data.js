import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import {
  useRouteMatch,
} from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import Nav from './Nav';
import pullData from '../redux/slices/dataSlice';

const Data = ({ name }) => {
  const { url } = useRouteMatch();
  const dispatch = useDispatch();

  const countryData = useSelector((state) => state.dataReducer.covidData);
  const covidStats = countryData[name];

  useEffect(async () => {
    if (!countryData.length) {
      dispatch(pullData(url));
    }
  }, []);

  return (
    <div className="country-container">
      <Nav title="Contry Details" backButton="< Back" />
      <div className="country-details">
        <h2>{covidStats?.name}</h2>
        <p>
          Source:
          {' '}
          {covidStats?.source}
        </p>
        <p>
          Deaths until today:
          {' '}
          {covidStats?.today_deaths}
        </p>
        <p>
          Deaths today:
          {' '}
          {covidStats?.today_new_deaths}
        </p>
        <p>
          Confirmed cases until today:
          {' '}
          {covidStats?.today_confirmed}
        </p>
        <p>
          Confirmed cases today:
          {' '}
          {covidStats?.today_new_confirmed}
        </p>
        <p>
          Open cases until today:
          {' '}
          {covidStats?.today_open_cases}
        </p>
        <p>
          Recovered cases until today:
          {' '}
          {covidStats?.today_recovered}
        </p>
        <p>
          New open cases:
          {' '}
          {covidStats?.today_new_open_cases}
        </p>
        <p>
          New recovered cases:
          {' '}
          {covidStats?.today_new_recovered}
        </p>
      </div>
    </div>
  );
};

Data.propTypes = ({
  name: PropTypes.string,
}).isRequired;

export default Data;
