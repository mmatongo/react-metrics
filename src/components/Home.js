import React, { useState } from 'react';
import {
  Link,
  useRouteMatch,
} from 'react-router-dom';
import { useSelector } from 'react-redux';
import Nav from './Nav';
import Search from './Search';

const Home = () => {
  const { url } = useRouteMatch();
  const [deaths, minDeaths] = useState('');

  const minDeathCount = (e) => {
    minDeaths(e.target.value);
  };

  const countries = useSelector((state) => state.covidReducer.covidCountries);
  const mappedList = countries.filter((country) => deaths <= country.deaths)
    .map((country) => (
      <Link
        key={country.name[0]}
        href="/#"
        to={`${url}${country.name[0].toLowerCase()}`}
        className="country-card"
      >
        <div>
          <div key={country.name[0]} id={country.name[0]} className="country-link">
            <h2 className="country-name">{country.name[0]}</h2>
            <p className="cases">
              TOTAL CASES CONFIRMED:
              {' '}
              {country.confirmed}
            </p>
            <p className="deaths">
              TOTAL DEATHS:
              {' '}
              {country.deaths}
            </p>
          </div>
        </div>
      </Link>
    ));

  return (
    <>
      <Nav title="COVID numbers per Country" />
      <Search value={deaths} handler={minDeathCount} />
      <div className="countries-container">
        { mappedList }
      </div>
    </>
  );
};

export default Home;
