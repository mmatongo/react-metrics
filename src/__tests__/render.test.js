import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../components/Home';
import store from '../containers/redux/configureStore';
import Nav from '../components/Nav';
import Search from '../components/Search';
import Data from '../components/Data';
import Back from '../components/Back';

describe('Test to see that all objects render correctly.', () => {
  test('Check if Home renders.', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Home /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Check if Nav renders.', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Nav /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Check if Search renders.', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Search /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Check if Data page renders.', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Data /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Check if Back button renders.', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><Back /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
