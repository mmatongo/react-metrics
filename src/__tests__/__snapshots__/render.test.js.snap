// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to see that all objects render correctly. Check if Back button renders. 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`Test to see that all objects render correctly. Check if Data page renders. 1`] = `
<div
  className="country-container"
>
  <div
    className="nav-container"
  >
    <div
      className="nav-contents"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        &lt; Back
      </button>
      <p>
        Contry Details
      </p>
      <div
        className="nav-icons"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="MicOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
          />
          <path
            d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="country-details"
  >
    <h2 />
    <p>
      Source:
       
    </p>
    <p>
      Deaths until today:
       
    </p>
    <p>
      Deaths today:
       
    </p>
    <p>
      Confirmed cases until today:
       
    </p>
    <p>
      Confirmed cases today:
       
    </p>
    <p>
      Open cases until today:
       
    </p>
    <p>
      Recovered cases until today:
       
    </p>
    <p>
      New open cases:
       
    </p>
    <p>
      New recovered cases:
       
    </p>
  </div>
</div>
`;

exports[`Test to see that all objects render correctly. Check if Home renders. 1`] = `
Array [
  <div
    className="nav-container"
  >
    <div
      className="nav-contents"
    >
      <button
        onClick={[Function]}
        type="button"
      />
      <p>
        COVID numbers per Country
      </p>
      <div
        className="nav-icons"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="MicOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
          />
          <path
            d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <div
    className="search"
  >
     Search by Death count: 
    <input
      id="deaths"
      name="deaths"
      onChange={[Function]}
      type="number"
      value=""
    />
  </div>,
  <div
    className="countries-container"
  />,
]
`;

exports[`Test to see that all objects render correctly. Check if Nav renders. 1`] = `
<div
  className="nav-container"
>
  <div
    className="nav-contents"
  >
    <button
      onClick={[Function]}
      type="button"
    />
    <p />
    <div
      className="nav-icons"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="MicOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
        />
        <path
          d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium nav-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="SettingsOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.09-.16-.26-.25-.44-.25-.06 0-.12.01-.17.03l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.06-.02-.12-.03-.18-.03-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.09.16.26.25.44.25.06 0 .12-.01.17-.03l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Test to see that all objects render correctly. Check if Search renders. 1`] = `
<div
  className="search"
>
   Search by Death count: 
  <input
    id="deaths"
    name="deaths"
    type="number"
  />
</div>
`;
